### ISSUE TITLE
Webdav link contains SUB-URI part twice

### ISSUE THREAD

**Issue by jonnywumic** [danmunn/redmine_dmsf#1088]
Install DMSF 2.4.2 with Bitnami Redmine 4.1.0 (with sub-uri /redmine by defult), can't access webdav by expected URL.
Expected: 
  http://192.168.122.163/redmine/dmsf/webdav/task-force
Actual: 
  http://192.168.122.163/redmine/redmine/dmsf/webdav/task-force

Redmine log
<pre>
Successful authentication for 'jonny.wu' from 192.168.122.196 at 2020-02-14 03:26:18 UTC
Redirected to http://192.168.122.163/redmine/
Completed 302 Found in 306ms (ActiveRecord: 182.1ms)
Started GET "/redmine/" for 192.168.122.196 at 2020-02-14 11:26:18 +0800
Processing by WelcomeController#index as HTML
  Current user: jonny.wu (id=3)
  Rendering welcome/index.html.erb within layouts/base
  Rendered collection of news/_news.html.erb [5 times] (25.4ms)
  Rendered welcome/index.html.erb within layouts/base (43.0ms)
  Rendered plugins/redmine_drawio/app/views/redmine_drawio/_macro_dialog.html.erb (0.4ms)
Completed 200 OK in 188ms (Views: 115.1ms | ActiveRecord: 12.5ms)

Started GET "/redmine/dmsf/webdav/task-force" for 192.168.122.196 at 2020-02-14 11:26:39 +0800

ActionController::RoutingError (No route matches [GET] "/dmsf/webdav/task-force"):

vendor/bundle/ruby/2.5.0/gems/actionpack-5.2.4.1/lib/action_dispatch/middleware/debug_exceptions.rb:65:in `call'
vendor/bundle/ruby/2.5.0/gems/actionpack-5.2.4.1/lib/action_dispatch/middleware/show_exceptions.rb:33:in `call'
vendor/bundle/ruby/2.5.0/gems/railties-5.2.4.1/lib/rails/rack/logger.rb:38:in `call_app'
vendor/bundle/ruby/2.5.0/gems/railties-5.2.4.1/lib/rails/rack/logger.rb:26:in `block in call'
vendor/bundle/ruby/2.5.0/gems/activesupport-5.2.4.1/lib/active_support/tagged_logging.rb:71:in `block in tagged'
vendor/bundle/ruby/2.5.0/gems/activesupport-5.2.4.1/lib/active_support/tagged_logging.rb:28:in `tagged'
vendor/bundle/ruby/2.5.0/gems/activesupport-5.2.4.1/lib/active_support/tagged_logging.rb:71:in `tagged'
vendor/bundle/ruby/2.5.0/gems/railties-5.2.4.1/lib/rails/rack/logger.rb:26:in `call'
vendor/bundle/ruby/2.5.0/gems/actionpack-5.2.4.1/lib/action_dispatch/middleware/remote_ip.rb:81:in `call'
vendor/bundle/ruby/2.5.0/gems/request_store-1.4.1/lib/request_store/middleware.rb:19:in `call'
vendor/bundle/ruby/2.5.0/gems/actionpack-5.2.4.1/lib/action_dispatch/middleware/request_id.rb:27:in `call'
vendor/bundle/ruby/2.5.0/gems/rack-2.0.8/lib/rack/method_override.rb:22:in `call'
vendor/bundle/ruby/2.5.0/gems/rack-2.0.8/lib/rack/runtime.rb:22:in `call'
vendor/bundle/ruby/2.5.0/gems/activesupport-5.2.4.1/lib/active_support/cache/strategy/local_cache_middleware.rb:29:in `call'
vendor/bundle/ruby/2.5.0/gems/actionpack-5.2.4.1/lib/action_dispatch/middleware/executor.rb:14:in `call'
vendor/bundle/ruby/2.5.0/gems/actionpack-5.2.4.1/lib/action_dispatch/middleware/static.rb:127:in `call'
vendor/bundle/ruby/2.5.0/gems/rack-2.0.8/lib/rack/content_length.rb:15:in `call'
vendor/bundle/ruby/2.5.0/gems/rack-2.0.8/lib/rack/sendfile.rb:111:in `call'
vendor/bundle/ruby/2.5.0/gems/railties-5.2.4.1/lib/rails/engine.rb:524:in `call'
vendor/bundle/ruby/2.5.0/gems/rack-2.0.8/lib/rack/urlmap.rb:68:in `block in call'
vendor/bundle/ruby/2.5.0/gems/rack-2.0.8/lib/rack/urlmap.rb:53:in `each'
vendor/bundle/ruby/2.5.0/gems/rack-2.0.8/lib/rack/urlmap.rb:53:in `call'
passenger (5.3.7) src/ruby_supportlib/phusion_passenger/rack/thread_handler_extension.rb:97:in `process_request'
passenger (5.3.7) src/ruby_supportlib/phusion_passenger/request_handler/thread_handler.rb:149:in `accept_and_process_next_request'
passenger (5.3.7) src/ruby_supportlib/phusion_passenger/request_handler/thread_handler.rb:110:in `main_loop'
passenger (5.3.7) src/ruby_supportlib/phusion_passenger/request_handler.rb:415:in `block (3 levels) in start_threads'
passenger (5.3.7) src/ruby_supportlib/phusion_passenger/utils.rb:113:in `block in create_thread_and_abort_on_exception'

Started GET "/redmine/redmine/dmsf/webdav/task-force" for 192.168.122.196 at 2020-02-14 11:26:47 +0800
Processing WebDAV request: /redmine/redmine/dmsf/webdav/task-force (for 192.168.122.196 at 2020-02-14 03:26:47 UTC) [GET]
Completed in: 147 ms | 200 [http://192.168.122.163/redmine/redmine/dmsf/webdav/task-force]

</pre>

Environment
<pre>
Environment:
  Redmine version                4.1.0.stable
  Ruby version                   2.5.7-p206 (2019-10-01) [x86_64-linux]
  Rails version                  5.2.4.1
  Environment                    production
  Database adapter               Mysql2
  Mailer queue                   ActiveJob::QueueAdapters::AsyncAdapter
  Mailer delivery                smtp
SCM:
  Git                            2.19.1
  Filesystem                     
Redmine plugins:
  redmine_agile                  1.5.2
  redmine_checklists             3.1.16
  redmine_dashboard              2.7.1
  redmine_dmsf                   2.4.2 devel
  redmine_drawio                 0.9.2
  redmine_issues_tree            0.0.14
</pre>

**Comment by picman** [comment]
The following option set?
https://github.com/danmunn/redmine_dmsf#installation-in-a-sub-uri

**Comment by jonnywumic** [comment]
> The following option set?
> https://github.com/danmunn/redmine_dmsf#installation-in-a-sub-uri

No such option in configuration files. 

Following is output when grep "redmine" in config/
<pre>
[root@redmine htdocs]# find config/ -type f -exec grep -H redmine {} \;
config/database.yml.example:  database: redmine
config/database.yml.example:  database: redmine_development
config/database.yml.example:  database: redmine_test
config/database.yml.example:#  database: redmine
config/database.yml.example:#  database: db/redmine.sqlite3
config/database.yml.example:#  database: redmine
config/environments/test.rb:  config.redmine_verify_sessions = false
config/application.rb:    config.redmine_verify_sessions = true
config/application.rb:    config.redmine_search_cache_store = :memory_store
config/application.rb:      :key => '_redmine_session',
config/settings.yml:  default: redmine@example.net
config/configuration.yml:  # http://www.redmine.org/projects/redmine/wiki/EmailConfiguration
config/configuration.yml:  # attachments_storage_path: /var/redmine/files
config/configuration.yml:  # attachments_storage_path: D:/redmine/files
config/configuration.yml:  # scm_stderr_log_file: /var/log/redmine_scm_stderr.log
config/configuration.yml:  # You can use `rake redmine:plugins:assets` to manually mirror assets
config/configuration.yml:  imagemagick_convert_command: '/opt/redmine-4.1.0-0/common/bin/convert'
config/initializers/30-redmine.rb:require 'redmine'
config/initializers/10-patches.rb:      redmine_default = [
config/initializers/10-patches.rb:      options[:default] = redmine_default + Array(options[:default])
config/initializers/10-patches.rb:  # See Defect #11545 (http://www.redmine.org/issues/11545) for details.

config/database.yml:  database: bitnami_redmine
config/database.yml:  socket: /opt/redmine-4.1.0-0/mysql/tmp/mysql.sock
config/database.yml:  database: redmine_development
config/database.yml:  database: redmine_test
config/database.yml:#  database: redmine
config/database.yml:#  database: db/redmine.sqlite3
config/database.yml:#  database: redmine

</pre>

**Comment by picman** [comment]
So add it as recommended in the README.

**Comment by jonnywumic** [comment]
> So add it as recommended in the README.

Added, the same symptom after restart redmine (apache, mysql)

<pre>
[root@redmine htdocs]# cat config/additional_environment.rb
# Copy this file to additional_environment.rb and add any statements
 config.relative_url_root = '/redmine'

Log output
<pre>
Started PROPFIND "/redmine/dmsf/webdav" for 10.99.0.196 at 2020-02-18 15:35:48 +0800

ActionController::RoutingError (No route matches [PROPFIND] "/dmsf/webdav"):

vendor/bundle/ruby/2.5.0/gems/actionpack-5.2.4.1/lib/action_dispatch/middleware/debug_exceptions.rb:65:in `call'
vendor/bundle/ruby/2.5.0/gems/actionpack-5.2.4.1/lib/action_dispatch/middleware/show_exceptions.rb:33:in `call'
vendor/bundle/ruby/2.5.0/gems/railties-5.2.4.1/lib/rails/rack/logger.rb:38:in `call_app'
vendor/bundle/ruby/2.5.0/gems/railties-5.2.4.1/lib/rails/rack/logger.rb:26:in `block in call'
vendor/bundle/ruby/2.5.0/gems/activesupport-5.2.4.1/lib/active_support/tagged_logging.rb:71:in `block in tagged'
vendor/bundle/ruby/2.5.0/gems/activesupport-5.2.4.1/lib/active_support/tagged_logging.rb:28:in `tagged'
vendor/bundle/ruby/2.5.0/gems/activesupport-5.2.4.1/lib/active_support/tagged_logging.rb:71:in `tagged'
vendor/bundle/ruby/2.5.0/gems/railties-5.2.4.1/lib/rails/rack/logger.rb:26:in `call'
vendor/bundle/ruby/2.5.0/gems/actionpack-5.2.4.1/lib/action_dispatch/middleware/remote_ip.rb:81:in `call'
vendor/bundle/ruby/2.5.0/gems/request_store-1.4.1/lib/request_store/middleware.rb:19:in `call'
vendor/bundle/ruby/2.5.0/gems/actionpack-5.2.4.1/lib/action_dispatch/middleware/request_id.rb:27:in `call'
vendor/bundle/ruby/2.5.0/gems/rack-2.0.8/lib/rack/method_override.rb:22:in `call'
vendor/bundle/ruby/2.5.0/gems/rack-2.0.8/lib/rack/runtime.rb:22:in `call'
vendor/bundle/ruby/2.5.0/gems/activesupport-5.2.4.1/lib/active_support/cache/strategy/local_cache_middleware.rb:29:in `call'
vendor/bundle/ruby/2.5.0/gems/actionpack-5.2.4.1/lib/action_dispatch/middleware/executor.rb:14:in `call'
vendor/bundle/ruby/2.5.0/gems/actionpack-5.2.4.1/lib/action_dispatch/middleware/static.rb:127:in `call'
vendor/bundle/ruby/2.5.0/gems/rack-2.0.8/lib/rack/content_length.rb:15:in `call'
vendor/bundle/ruby/2.5.0/gems/rack-2.0.8/lib/rack/sendfile.rb:111:in `call'
vendor/bundle/ruby/2.5.0/gems/railties-5.2.4.1/lib/rails/engine.rb:524:in `call'
vendor/bundle/ruby/2.5.0/gems/rack-2.0.8/lib/rack/urlmap.rb:68:in `block in call'
vendor/bundle/ruby/2.5.0/gems/rack-2.0.8/lib/rack/urlmap.rb:53:in `each'
vendor/bundle/ruby/2.5.0/gems/rack-2.0.8/lib/rack/urlmap.rb:53:in `call'
passenger (5.3.7) src/ruby_supportlib/phusion_passenger/rack/thread_handler_extension.rb:97:in `process_request'
passenger (5.3.7) src/ruby_supportlib/phusion_passenger/request_handler/thread_handler.rb:157:in `accept_and_process_next_request'
passenger (5.3.7) src/ruby_supportlib/phusion_passenger/request_handler/thread_handler.rb:110:in `main_loop'
passenger (5.3.7) src/ruby_supportlib/phusion_passenger/request_handler.rb:415:in `block (3 levels) in start_threads'
passenger (5.3.7) src/ruby_supportlib/phusion_passenger/utils.rb:113:in `block in create_thread_and_abort_on_exception'
Started GET "/redmine/dmsf/webdav" for 10.99.0.196 at 2020-02-18 15:35:48 +0800

ActionController::RoutingError (No route matches [GET] "/dmsf/webdav"):

vendor/bundle/ruby/2.5.0/gems/actionpack-5.2.4.1/lib/action_dispatch/middleware/debug_exceptions.rb:65:in `call'
vendor/bundle/ruby/2.5.0/gems/actionpack-5.2.4.1/lib/action_dispatch/middleware/show_exceptions.rb:33:in `call'
vendor/bundle/ruby/2.5.0/gems/railties-5.2.4.1/lib/rails/rack/logger.rb:38:in `call_app'
vendor/bundle/ruby/2.5.0/gems/railties-5.2.4.1/lib/rails/rack/logger.rb:26:in `block in call'
vendor/bundle/ruby/2.5.0/gems/activesupport-5.2.4.1/lib/active_support/tagged_logging.rb:71:in `block in tagged'
vendor/bundle/ruby/2.5.0/gems/activesupport-5.2.4.1/lib/active_support/tagged_logging.rb:28:in `tagged'
vendor/bundle/ruby/2.5.0/gems/activesupport-5.2.4.1/lib/active_support/tagged_logging.rb:71:in `tagged'
vendor/bundle/ruby/2.5.0/gems/railties-5.2.4.1/lib/rails/rack/logger.rb:26:in `call'
vendor/bundle/ruby/2.5.0/gems/actionpack-5.2.4.1/lib/action_dispatch/middleware/remote_ip.rb:81:in `call'
vendor/bundle/ruby/2.5.0/gems/request_store-1.4.1/lib/request_store/middleware.rb:19:in `call'
vendor/bundle/ruby/2.5.0/gems/actionpack-5.2.4.1/lib/action_dispatch/middleware/request_id.rb:27:in `call'
vendor/bundle/ruby/2.5.0/gems/rack-2.0.8/lib/rack/method_override.rb:22:in `call'
vendor/bundle/ruby/2.5.0/gems/rack-2.0.8/lib/rack/runtime.rb:22:in `call'
vendor/bundle/ruby/2.5.0/gems/activesupport-5.2.4.1/lib/active_support/cache/strategy/local_cache_middleware.rb:29:in `call'
vendor/bundle/ruby/2.5.0/gems/actionpack-5.2.4.1/lib/action_dispatch/middleware/executor.rb:14:in `call'
vendor/bundle/ruby/2.5.0/gems/actionpack-5.2.4.1/lib/action_dispatch/middleware/static.rb:127:in `call'
vendor/bundle/ruby/2.5.0/gems/rack-2.0.8/lib/rack/content_length.rb:15:in `call'
vendor/bundle/ruby/2.5.0/gems/rack-2.0.8/lib/rack/sendfile.rb:111:in `call'
vendor/bundle/ruby/2.5.0/gems/railties-5.2.4.1/lib/rails/engine.rb:524:in `call'
vendor/bundle/ruby/2.5.0/gems/rack-2.0.8/lib/rack/urlmap.rb:68:in `block in call'
vendor/bundle/ruby/2.5.0/gems/rack-2.0.8/lib/rack/urlmap.rb:53:in `each'
vendor/bundle/ruby/2.5.0/gems/rack-2.0.8/lib/rack/urlmap.rb:53:in `call'
passenger (5.3.7) src/ruby_supportlib/phusion_passenger/rack/thread_handler_extension.rb:97:in `process_request'
passenger (5.3.7) src/ruby_supportlib/phusion_passenger/request_handler/thread_handler.rb:149:in `accept_and_process_next_request'
passenger (5.3.7) src/ruby_supportlib/phusion_passenger/request_handler/thread_handler.rb:110:in `main_loop'
passenger (5.3.7) src/ruby_supportlib/phusion_passenger/request_handler.rb:415:in `block (3 levels) in start_threads'
passenger (5.3.7) src/ruby_supportlib/phusion_passenger/utils.rb:113:in `block in create_thread_and_abort_on_exception'

</pre>
</pre>

**Comment by picman** [comment]
Hm :-(
I think that the crutial part is in _lib/redmine_dmsf/webdav/custom_middleware.rb_

Could you add the following line into the code and post what is in the _production.log_ then?

```ruby
29 def initialize(app)
30        @rails_app = app
31        path = File.join(Redmine::Utils::relative_url_root, %w(dmsf webdav))
32 ++   Rails.logger.info ">>> #{path}"
        @dav_app = Rack::Builder.new{
...
````

**Comment by jonnywumic** [comment]
> Hm :-(
> I think that the crutial part is in _lib/redmine_dmsf/webdav/custom_middleware.rb_
> 
> Could you add the following line into the code and post what is in the _production.log_ then?
> 
> ```ruby
> 29 def initialize(app)
> 30        @rails_app = app
> 31        path = File.join(Redmine::Utils::relative_url_root, %w(dmsf webdav))
> 32 ++   Rails.logger.info ">>> #{path}"
>         @dav_app = Rack::Builder.new{
> ...
> ```


production.log after add 
<pre>
Creating scope :system. Overwriting existing method Enumeration.system.
Creating scope :sorted. Overwriting existing method Group.sorted.
Creating scope :sorted. Overwriting existing method User.sorted.
Creating scope :having_mail. Overwriting existing method User.having_mail.
Creating scope :visible. Overwriting existing method AgileQuery.visible.
>>> /redmine/dmsf/webdav
DEPRECATION WARNING: Using a dynamic :action segment in a route is deprecated and will be removed in Rails 6.0. (called from instance_eval at /opt/redmine-4.1.0-0/apps/redmine/htdocs/config/routes.rb:370)
DEPRECATION WARNING: Using a dynamic :action segment in a route is deprecated and will be removed in Rails 6.0. (called from instance_eval at /opt/redmine-4.1.0-0/apps/redmine/htdocs/config/routes.rb:370)
DEPRECATION WARNING: Using a dynamic :action segment in a route is deprecated and will be removed in Rails 6.0. (called from instance_eval at /opt/redmine-4.1.0-0/apps/redmine/htdocs/config/routes.rb:370)
DEPRECATION WARNING: Using a dynamic :action segment in a route is deprecated and will be removed in Rails 6.0. (called from instance_eval at /opt/redmine-4.1.0-0/apps/redmine/htdocs/config/routes.rb:370)
DEPRECATION WARNING: Using a dynamic :action segment in a route is deprecated and will be removed in Rails 6.0. (called from instance_eval at /opt/redmine-4.1.0-0/apps/redmine/htdocs/config/routes.rb:370)
DEPRECATION WARNING: Using a dynamic :action segment in a route is deprecated and will be removed in Rails 6.0. (called from instance_eval at /opt/redmine-4.1.0-0/apps/redmine/htdocs/config/routes.rb:370)
DEPRECATION WARNING: Using a dynamic :action segment in a route is deprecated and will be removed in Rails 6.0. (called from instance_eval at /opt/redmine-4.1.0-0/apps/redmine/htdocs/config/routes.rb:370)
Started HEAD "/redmine" for 127.0.0.1 at 2020-02-21 16:11:06 +0800
Processing by WelcomeController#index as HTML
  Current user: anonymous
Redirected to http://127.0.0.1/redmine/login?back_url=http%3A%2F%2F127.0.0.1%2Fredmine%2F
Filter chain halted as :check_if_login_required rendered or redirected
Completed 302 Found in 104ms (ActiveRecord: 8.4ms)
</pre>

**Comment by picman** [comment]
It looks good. Then, WebDAV should be available at https://yourdomain.com/redmine/dmsf/webdav.

**Comment by jonnywumic** [comment]
No, It still look for "/redmine/redmine/dmsf/webdav"

<pre>
Creating scope :system. Overwriting existing method Enumeration.system.
Creating scope :sorted. Overwriting existing method Group.sorted.
Creating scope :sorted. Overwriting existing method User.sorted.
Creating scope :having_mail. Overwriting existing method User.having_mail.
Creating scope :visible. Overwriting existing method AgileQuery.visible.
>>> /redmine/dmsf/webdav
DEPRECATION WARNING: Using a dynamic :action segment in a route is deprecated and will be removed in Rails 6.0. (called from instance_eval at /opt/redmine-4.1.0-0/apps/redmine/htdocs/config/routes.rb:370)
DEPRECATION WARNING: Using a dynamic :action segment in a route is deprecated and will be removed in Rails 6.0. (called from instance_eval at /opt/redmine-4.1.0-0/apps/redmine/htdocs/config/routes.rb:370)
DEPRECATION WARNING: Using a dynamic :action segment in a route is deprecated and will be removed in Rails 6.0. (called from instance_eval at /opt/redmine-4.1.0-0/apps/redmine/htdocs/config/routes.rb:370)
DEPRECATION WARNING: Using a dynamic :action segment in a route is deprecated and will be removed in Rails 6.0. (called from instance_eval at /opt/redmine-4.1.0-0/apps/redmine/htdocs/config/routes.rb:370)
DEPRECATION WARNING: Using a dynamic :action segment in a route is deprecated and will be removed in Rails 6.0. (called from instance_eval at /opt/redmine-4.1.0-0/apps/redmine/htdocs/config/routes.rb:370)
DEPRECATION WARNING: Using a dynamic :action segment in a route is deprecated and will be removed in Rails 6.0. (called from instance_eval at /opt/redmine-4.1.0-0/apps/redmine/htdocs/config/routes.rb:370)
DEPRECATION WARNING: Using a dynamic :action segment in a route is deprecated and will be removed in Rails 6.0. (called from instance_eval at /opt/redmine-4.1.0-0/apps/redmine/htdocs/config/routes.rb:370)
Started HEAD "/redmine" for 127.0.0.1 at 2020-02-21 16:52:16 +0800
Processing by WelcomeController#index as HTML
  Current user: anonymous
Redirected to http://127.0.0.1/redmine/login?back_url=http%3A%2F%2F127.0.0.1%2Fredmine%2F
Filter chain halted as :check_if_login_required rendered or redirected
Completed 302 Found in 95ms (ActiveRecord: 7.1ms)

Started GET "/redmine/dmsf/webdav" for 192.168.0.196 at 2020-02-21 16:52:34 +0800

ActionController::RoutingError (No route matches [GET] "/dmsf/webdav"):

vendor/bundle/ruby/2.5.0/gems/actionpack-5.2.4.1/lib/action_dispatch/middleware/debug_exceptions.rb:65:in `call'
vendor/bundle/ruby/2.5.0/gems/actionpack-5.2.4.1/lib/action_dispatch/middleware/show_exceptions.rb:33:in `call'
vendor/bundle/ruby/2.5.0/gems/railties-5.2.4.1/lib/rails/rack/logger.rb:38:in `call_app'
vendor/bundle/ruby/2.5.0/gems/railties-5.2.4.1/lib/rails/rack/logger.rb:26:in `block in call'
vendor/bundle/ruby/2.5.0/gems/activesupport-5.2.4.1/lib/active_support/tagged_logging.rb:71:in `block in tagged'
vendor/bundle/ruby/2.5.0/gems/activesupport-5.2.4.1/lib/active_support/tagged_logging.rb:28:in `tagged'
vendor/bundle/ruby/2.5.0/gems/activesupport-5.2.4.1/lib/active_support/tagged_logging.rb:71:in `tagged'
vendor/bundle/ruby/2.5.0/gems/railties-5.2.4.1/lib/rails/rack/logger.rb:26:in `call'
vendor/bundle/ruby/2.5.0/gems/actionpack-5.2.4.1/lib/action_dispatch/middleware/remote_ip.rb:81:in `call'
vendor/bundle/ruby/2.5.0/gems/request_store-1.4.1/lib/request_store/middleware.rb:19:in `call'
vendor/bundle/ruby/2.5.0/gems/actionpack-5.2.4.1/lib/action_dispatch/middleware/request_id.rb:27:in `call'
vendor/bundle/ruby/2.5.0/gems/rack-2.0.8/lib/rack/method_override.rb:22:in `call'
vendor/bundle/ruby/2.5.0/gems/rack-2.0.8/lib/rack/runtime.rb:22:in `call'
vendor/bundle/ruby/2.5.0/gems/activesupport-5.2.4.1/lib/active_support/cache/strategy/local_cache_middleware.rb:29:in `call'
vendor/bundle/ruby/2.5.0/gems/actionpack-5.2.4.1/lib/action_dispatch/middleware/executor.rb:14:in `call'
vendor/bundle/ruby/2.5.0/gems/actionpack-5.2.4.1/lib/action_dispatch/middleware/static.rb:127:in `call'
vendor/bundle/ruby/2.5.0/gems/rack-2.0.8/lib/rack/content_length.rb:15:in `call'
vendor/bundle/ruby/2.5.0/gems/rack-2.0.8/lib/rack/sendfile.rb:111:in `call'
vendor/bundle/ruby/2.5.0/gems/railties-5.2.4.1/lib/rails/engine.rb:524:in `call'
vendor/bundle/ruby/2.5.0/gems/rack-2.0.8/lib/rack/urlmap.rb:68:in `block in call'
vendor/bundle/ruby/2.5.0/gems/rack-2.0.8/lib/rack/urlmap.rb:53:in `each'
vendor/bundle/ruby/2.5.0/gems/rack-2.0.8/lib/rack/urlmap.rb:53:in `call'
passenger (5.3.7) src/ruby_supportlib/phusion_passenger/rack/thread_handler_extension.rb:97:in `process_request'
passenger (5.3.7) src/ruby_supportlib/phusion_passenger/request_handler/thread_handler.rb:149:in `accept_and_process_next_request'
passenger (5.3.7) src/ruby_supportlib/phusion_passenger/request_handler/thread_handler.rb:110:in `main_loop'
passenger (5.3.7) src/ruby_supportlib/phusion_passenger/request_handler.rb:415:in `block (3 levels) in start_threads'
passenger (5.3.7) src/ruby_supportlib/phusion_passenger/utils.rb:113:in `block in create_thread_and_abort_on_exception'


Started GET "/redmine/redmine/dmsf/webdav" for 192.168.0.196 at 2020-02-21 16:52:42 +0800
Processing WebDAV request: /redmine/redmine/dmsf/webdav/ (for 192.168.0.196 at 2020-02-21 08:52:42 UTC) [GET]
Completed in: 30 ms | 401 [http://192.168.241.163/redmine/redmine/dmsf/webdav/]
Started GET "/redmine/redmine/dmsf/webdav" for 192.168.0.196 at 2020-02-21 16:52:49 +0800
Processing WebDAV request: /redmine/redmine/dmsf/webdav/ (for 192.168.0.196 at 2020-02-21 08:52:49 UTC) [GET]
Scoped order is ignored, it's forced to be batch order.
Completed in: 1186 ms | 200 [http://192.168.241.163/redmine/redmine/dmsf/webdav/]
</pre>

**Comment by willuhn** [comment]
I had exactly the same error (also using a sub uri "/redmine"). https://github.com/danmunn/redmine_dmsf/issues/1088 fixed it for me. "config.relative_url_root = '/redmine'" according to https://github.com/danmunn/redmine_dmsf#installation-in-a-sub-uri is no longer required after that change (at least for my installation).

**Comment by enzote84** [comment]
Hi!
I have the same problem.
After creating config/additional_environment.rb with config.relative_url_root = '/redmine', the problem persists.
It works if I use doman/redmine/redmine/dmsf/webdav/project.

**Comment by parrier82** [comment]
I'm using the below system. 

  Redmine version                4.1.1.stable.19980
  Ruby version                     2.6.5-p114 (2019-10-01) [x86_64-linux]
  Rails version                      5.2.4.2
  redmine_dmsf                    2.4.4

When I tried to save drawing in wiki by using draw.io plugin, Redmine couldn't save the drawing. And It shows 
"Error saving diagram : Make sure webdav capabilities of DMSF module is enabled". 

Unfortunately, I already activated Webdav function and choosed read/write function in DMSF plugin setting. And I also tried to add [ config.relative_url_root = '/redmine' ] to config/additional_environment.rb. But there is nothing to change. 

So, I tried to access "mydomain/redmine/dmsf/webdav". It occurred fail. But When I tried to access "mydomain/**redmine/redmine**/dmsf/webdav", It worked. 

My default address to access redmine is "**mydomain/redmine**". Is it able to make problem?
